Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -225,10 +228,11 @@ protected void SendEventInternal(Type invokeClass, string eventName, NetworkWrit
                 netId = netId,
                 componentIndex = ComponentIndex,
                 functionHash = GetMethodHash(invokeClass, eventName), // type+func so Inventory.RpcUse != Equipment.RpcUse
-                payload = new ArraySegment<byte>(writer.ToArray()) // segment to avoid reader allocations
+                payload = writer.ToArraySegment() // segment to avoid reader allocations
             };
 
             NetworkServer.SendToReady(netIdentity,message, channelId);
+            NetworkWriterPool.Recycle(writer);
```

Comment:
wouldn't it be cleaner to recycle in the weaver?

Reply:
Perhaps,   I avoid weaver code like the plague,  but indeed it is much cleaner to do it in the weaver.